app/src/main/java/com/duckduckgo/app/browser/BrowserWebViewClient.kt

```diff
@@ -166,6 +168,57 @@ class BrowserWebViewClient @Inject constructor(
         super.onReceivedSslError(view, handler, error)
     }
 
+    @UiThread
+    override fun onReceivedHttpAuthRequest(view: WebView?, handler: HttpAuthHandler?, host: String?, realm: String?) {
+        Timber.v("onReceivedHttpAuthRequest ${view?.url} $realm, $host,  $currentUrl")
+
+        if (handler != null) {
+            Timber.v("onReceivedHttpAuthRequest - useHttpAuthUsernamePassword [${handler.useHttpAuthUsernamePassword()}]")
+            if (handler.useHttpAuthUsernamePassword()) {
+                val webViewDatabase = WebViewDatabase.getInstance(view?.context)
+
+                val credentials = if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.O) {
+                    webViewDatabase.getHttpAuthUsernamePassword(host.orEmpty(), realm.orEmpty())
+                } else {
+                    @Suppress("DEPRECATION")
+                    view?.getHttpAuthUsernamePassword(host.orEmpty(), realm.orEmpty())
+                }
+
+                credentials?.let {
```

Comment:
we tend to only use `?.let` for when we don't care about the `null` case (though i'm sure there are plenty of cases where we don't stick to that convention)

however, here, i think it would be clearer to give with an `if/else`

Reply:
👍 That seems like a reasonable convention.